lib/browser/api/browser-window.js

```diff
@@ -72,44 +72,84 @@ BrowserWindow.prototype._init = function () {
       return this.webContents.devToolsWebContents;
     }
   });
+};
 
-  // Properties
+// Properties
 
-  Object.defineProperty(this, 'autoHideMenuBar', {
-    get: () => this.isMenuBarAutoHide(),
-    set: (autoHide) => this.setAutoHideMenuBar(autoHide)
-  });
+Object.defineProperty(BrowserWindow.prototype, 'autoHideMenuBar', {
```

Comment:
Can you move the properties to `TopLevelWindow`? These are essentially properties of `TopLevelWindow` and inherited by `BrowserWindow`, once we make `TopLevelWindow` public per https://github.com/electron/governance/pull/254 we will have to move the properties to it anyway.